src/io/mod.rs

```diff
@@ -313,6 +331,27 @@ mod tests {
         assert_ne!(text.metadata().unwrap().len(), gzipped.metadata().unwrap().len());
     }
 
+    #[test]
+    fn test_reading_and_writing_zstd_files() {
+        let lines = vec!["foo", "bar", "baz"];
+        let tempdir = TempDir::new().unwrap();
+        let text = tempdir.path().join("text.txt");
+        let zstd_compressed = tempdir.path().join("zstd_compressed.txt.zst");
```

Comment:
consider a unit test, or an assert here, to validate that `is_zstd_path` returns false for `text` and true for `zstd_compressed`.  I.e. what if the bug was that it just gzipped regardless?

Reply:
Done